src/daft-minhash/src/minhash.rs

```diff
@@ -0,0 +1,215 @@
+use std::{
+    ops::{Add, BitAnd, Mul, Rem, Shr},
+    simd::{cmp::SimdOrd, Simd},
+};
+
+use common_error::DaftResult;
+use mur3::murmurhash3_x86_32;
+
+// which SIMD to use
+const SIMD_LANES: usize = 8;
+type S = Simd<u64, SIMD_LANES>;
+
+const MERSENNE_PRIME: u64 = (1 << 61) - 1;
+// const MERSENNE_SIMD: S = S::from_array([MERSENNE_PRIME; SIMD_LANES]);
+// const MERSENNE_EXP_SIMD: S = S::from_array([61; SIMD_LANES]);
+const MAX_HASH: u64 = 0xffffffff;
+const MAX_HASH_SIMD: S = S::from_array([MAX_HASH; SIMD_LANES]);
+
+#[inline(always)]
+fn fast_simd_rem(x: S) -> S {
+    (x + x.shr(61)).bitand(MAX_HASH_SIMD)
```

Comment:
I wasn't convinced that this works so I put it in a theorem prover and looks like it is true only if `x < u32::MAX`, we should put a comment here for that and a `debug_assert`

<img width="652" alt="image" src="https://github.com/Eventual-Inc/Daft/assets/2550285/ba659232-03cf-4871-a65e-484e4d930777">

Reply:
We are definitely passing in values greater than `u32::MAX`, as the input is the product of two 32-bit integers. My claim is that it fails with probability <= 2^-58 over uniform inputs, and this is acceptable enough for hashing.